### ISSUE TITLE
Allow singleton mode to use Discogs IDs

### ISSUE THREAD

**Issue by arawden** [beetbox/beets#3574]
### Use case
I have a techno library collection which consits of some folders, with varying levels of specificity, and a large collection of individual files. I don't analyze the entire collection at once, rather I add folders where I think they belong and then import them into beets afterwards. 

I would like to be able to import the set of individual files to my library with something like `beet import -s *.mp3`, but when I do, beets has a hard time matching the tracks to the correct versions.

Many of these songs exist as releases on albums which are available from Discogs. Consider the file `Anetha - Endless Sea.mp3`, which is part of [this release](https://www.discogs.com/Various-Urban-Utopia/release/10047361).

I wish to be able to provide the Discogs ID to beets while in singleton mode and have it check the tracks which are part of the release, in order to properly tag the files. 

### Solution
I think the simplest solution would be to match the artist and track information against the tracks in the release. I'm not sure what the response data looks like from Discogs, but I would hazard a guess that right now beets doesn't really respect the singleton mode and checks the track against the album.

 I would be happy to work on this myself, I would just need some guidance. 

### Alternatives
I haven't looked into alternatives super far, but I saw someone else had commented on #2544 about a month ago, so I suspect the options are limited.

**Comment by sampsyo** [comment]
Hello! Any chance you could expand a little bit? I'm not entirely sure what you mean by this:

> I think the simplest solution would be to match the artist and track information against the tracks in the release.

The beets matching process already works by comparing the artist and track against tracks from Discogs (or any other metadata source).

I think perhaps you mean *searching* Discogs in singleton mode, as discussed in https://github.com/beetbox/beets/issues/2544#issuecomment-611514884? Indeed; I don't think that would be too hard. You can get started by looking at how the Discogs plugin currently works in its album-based search mode, and perhaps at how other metadata sources (e.g., Beatport) work in singleton mode.

**Comment by arawden** [comment]
Oh, I had not really considered that. When I wrote:
> I think the simplest solution would be to match the artist and track information against the tracks in the release.  

I meant that when I provide beets with an ID to a Discogs release, the program would examine the provided release for matching aritst/track metadat.

> The beets matching process already works by comparing the artist and track against tracks from Discogs (or any other metadata source).

My understanding is beets, when run outside of singleton mode, uses the name of the directory to match against releases, i.e. if I had a folder named `Various - Urban Utopia`, it would know to examine that release and find the Anetha track I used in my example. When run in singleton mode, however, beets gets confused. In my mind, providing the ID to a release while in singleton mode would clarify where beets should look for metadata information.

I hope that explains my thoughts a bit.
